ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+field in the [Translator](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L48)
+struct.
+
+Other per-component mappings to Helm values.yaml are expressed in the
+[ComponentMaps](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/translate/translate.go#L64)
+struct.
+
+### Validations
+
+Both the new API and old Helm API are validated. The new API is validated through a table of validation rules in
+[pkg/validate/validate.go](https://github.com/istio/operator/blob/master/pkg/validate/validate.go). These rules
+refer to the Go struct path schema and hence have names with a capitalized first letter.
+The old Helm values.yaml API is validated in
+[validate_values.go](https://github.com/istio/operator/blob/master/pkg/validate/validate_values.go)
+and refer to the values.yaml data paths. Hence, these rules have names with a lower case first letter.
+
+## K8s controller
+
+WIP.
+
+## Manifest creation
+
+Manifest rendering is a multi-step process, shown in the figure below.
+![rendering process](images/operator_render_flow.png)
+The example in the figure shows the rendering being triggered by a CLI `iop` command with a IstioControlPlane CR passed
+to it from a file; however, the same rendering steps would occur when an in-cluster CR is updated and the controller
+acts upon it to generate a new manifest to apply to the cluster.
+Note that both the charts and configuration profiles can come from three different sources: compiled-in, local
+filesystem, or URL. The source may be selected independently for the charts and profiles.
+The different steps in creating the manifest are as follows:
+
+1. The user CR (my_custom.yaml) selects a configuration profile. If no profile is selected, the
+[default profile](https://github.com/istio/operator/blob/master/data/profiles/default.yaml) is used. Each profile
+is defined as a set of defaults for IstioControlPlane, for both the new API (K8s settings, namespaces and enablement)
+and the old Helm values (Istio behavior configuration).
+1. The new API fields defined in the user CR override any values defined in the configuration profile CR.  The
+resulting CR is converted to Helm values.yaml format and passed to the next step.
+1. Part of the configuration profile contains settings in the old API values.yaml schema format. User overrides of
+these fields are applied and merged with the output of this step. The result of this step is a merge of configuration
+profile defaults and user overlays, all expressed in Helm values.yaml format. This final values.yaml configuration
+is passed to the Helm rendering library and used to render the charts. The rendered manifests are passed to the next
+step.
+1. Overlays in the user CR are applied to the rendered manifests. No values are ever defined in configuration profile
+CRs at this layer, so no merge is performed in this step.
```

Comment:
this whole section refers to new api and old api.  Instead of referring to them by age refer to them by some other form.  new and old are hard to parse (what functioanlity is in the old what is in the new).  This is a naming problem :)

Reply:
Done.

Reply:
Please use "IstioControlPlaneSpec API" consistently instead of "new API". There are still a few occurrences of "new API" in the doc.

Reply:
Also, there are still occurrences of "old" in the doc. To me, this implies that this functionality will get deprecated at some point. Is that correct? What is the plan to replace Helm values.yaml in the operator? It seems like it's still an important part of the manifest generation. If this won't get deprecated, don't call it "old".